### PR TITLE
Themes: Disable filtering for Jetpack sites; enable deletion only for existing themes

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR disables the unnecessary filtering of themes for Jetpack sites - this was causing any wpcom themes that exist on the site to not be added to the list of themes on the site in Redux at all.

We also enable deletion only for themes that are found on a site; if a theme isn't found, the delete button shouldn't appear at all.

An alternative to #51277 and #51300.

#### Testing instructions

* Go to `/themes/:site` where `:site` is a Jetpack site.
* Activate "Twentytwenty"
* Activate "Twentytwentyone"
* Verify you can delete Twentytwenty successfully.
* Re-check and verify you can no longer delete Twentytwenty now (the deletion button is no longer visible).
* Verify you still can't delete "Twentytwentyone".
* Verify that if you activate another theme, you can now delete "Twentytwentyone"
* Verify the themes list and deletion of other themes works the same as before.

### CODE REVIEW

File: client/state/themes/actions/receive-themes.js [hunk 1]
```diff
@@ -8,8 +8,7 @@ import { filter } from 'lodash';
  */
 import { THEMES_REQUEST_SUCCESS } from 'calypso/state/themes/action-types';
 import { isJetpackSite } from 'calypso/state/sites/selectors';
-import { shouldFilterWpcomThemes } from 'calypso/state/themes/selectors';
-import { isThemeFromWpcom, isThemeMatchingQuery } from 'calypso/state/themes/utils';
```
Comment (jsnajdr) [inline client/state/themes/actions/receive-themes.js:12]: Both `shouldFilterWpcomThemes` and `isThemeFromWpcom` lost their last usage and can be removed.

**Review by jsnajdr — APPROVED** [review]
I can confirm that I can now delete only themes that are actually uploaded on the site. Both from the "Recommended" list and also from the "Advanced" list.

The `shouldFilterWpcomThemes` filtering was added by @seear in #11516 but I don't understand what was the problem it was fixing.

The current UI for Jetpack Themes is still very confusing. There are three lists:
- recommended
- after expanding a "show all" button, I see:
  -  search box
  - "uploaded" themes without any label or distinction from the next list
  - "advanced" searchable list of all wpcom themes including the "not recommended" ones

This UI probably works for the use case where I want to find a new theme for my site and install it. I.e., something like "shopping in a theme store".

But if I wanted to see themes actually uploaded on my Jetpack site and delete one of them, I'm afraid I wouldn't be able to figure out that. For the "managing installed themes on my site", the UI doesn't work.

File: client/state/themes/actions/receive-themes.js (hunk 1 above)
Comment (tyxla) [reply client/state/themes/actions/receive-themes.js:12]: Good call 👍  Removed in acca3068c40643930ee3f56fa0bc6fb5013795dd

**Comment by tyxla** [comment]
Thanks for the thorough reviews and insights on this @jsnajdr 😍

**Pull request merged**